src/Compilers/CSharp/Portable/Binder/Binder_Lookup.cs

```diff
@@ -1371,7 +1371,7 @@ FileIdentifier getFileIdentifierForFileTypes()
                     if (binder is BuckStopsHereBinder lastBinder)
                     {
                         // we never expect to bind a file type in a context where the BuckStopsHereBinder lacks an AssociatedFileIdentifier
-                        return lastBinder.AssociatedFileIdentifier.Value;
+                        return lastBinder.AssociatedFileIdentifier;
```

Comment:
Let's insert `?? throw ExceptionUtilities.Unreachable()` here so we preserve the null test being performed by the original `.Value` access.